tutorials/shading/shading_reference/shading_language.rst

```diff
@@ -187,6 +187,65 @@ Keep in mind that some architectures (mainly mobile) benefit a lot from this, bu
 to find out more. In all honesty though, mobile drivers are buggy, so, to stay out of trouble, make simple 
 shaders without specifying precision unless you *really* need to.
 
+Arrays
+------
+
+Arrays are containers for the multiple variables of similar type. In GLSL they can be declared local to function, global space or uniform.
+Note: At the time of 3.2 only local arrays has been implemented.
+
+Local arrays
+~~~~~~~~~~~~
+
+Local arrays are declared in the functions. They can use all of the allowed datatypes, except the samplers.
+Declaration of the array is follow the C syntax : 'typename + identifier + [array size]'.
+
+.. code-block:: glsl
+
+    void fragment() {
+        float arr[3];
+    }
+
+They can be initialized at the beginning like:
+
+.. code-block:: glsl
+
+      float arr[3] = float[3] (1.0, 0.5, 0.0); // first init form
+
+      vec2 arr_v2[2] = { vec2(0.0, 0.0), vec2(1.0, 1.0) }; // second form
+
+      bool bvec_arr[] = { false, false, true, true }; // third form - size is defined automatically from the argument count
+
+      // Note: The last two forms are syntax sugar - these forms are not presented in GLSL.
+
+You can declare multiple arrays (even with different sizes) in one line like:
+
+.. code-block:: glsl
+
+      float a[3] = float[3] (1.0, 0.5, 0.0), b[2] = {1.0, 0.5}, c[] = { 0.7 }, d = 0.0, e[5];
```

Comment:
Maybe this could be split over several lines for readability (if possible) :slightly_smiling_face: